Cython/Compiler/ExprNodes.py

```diff
@@ -3420,6 +3422,13 @@ def generate_result_code(self, code):
             code.globalstate.use_utility_code(
                 UtilityCode.load_cached("PyObjectFormatSimple", "StringTools.c"))
 
+        if self.expr_text:
+            expr_text = self.expr_text + '='
+            value_result = 'PyNumber_Add(PyUnicode_FromStringAndSize("%s", %s), %s)' % (
```

Comment:
I'm pretty sure this is leaking the result of `PyUnicode_FromStringAndSize`, and probably the result of `PyNumber_Add` too.

I think this might have to be a new section in `Cython/Utility/Stringtools.c` similar to `PyObjectFormatAndDecref` etc. But this is the first time I've looked at the fstring implementation so I'm not complete confident about the best way to do it.

Reply:
An alternative approach that might be worth a go is to generate a different set of nodes (probably at the `analyse_ expressions` stage). Something like:

```
# in analyse_expressions for FormattedValueNode:
if self.expr_text:
    return AddNode(self.pos, lhs=UnicodeConstantNode(self.pos, value=self.expr_text+"="),
                                       rhs=<copy of self but without self.expr_text>).analyse_expressions(env)
```

The advantage of doing something like that is that Cython's code generation should take care of most of the details. Note that the code above is a very vague outline...

Reply:
Will have a try with that approach :+1: 

Btw, mind canceling all the old builds? They are taking for ever to complete:
https://travis-ci.org/github/cython/cython/builds/748578596
https://travis-ci.org/github/cython/cython/builds/748556005

Thanks again!

Reply:
> Btw, mind canceling all the old builds?

I don't have the power I'm afraid.

Reply:
Oh, thought you did, nvm then :+1:

Reply:
Implemented :smile: